OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -451,6 +453,17 @@ void ConnectionStateChanged(OrderManager om)
 			MiniYaml yaml;
 			if (logicArgs.TryGetValue("ChatLineSound", out yaml))
 				chatLineSound = yaml.Value;
+
+			discordState = new DiscordState
+			{
+				State = "In Lobby",
+				Details = map.Title,
+				StartTime = DateTime.UtcNow
+			};
+
+			var discordService = Game.ModData.Manifest.Get<DiscordService>();
```

Comment:
We could save some duplication by defining a static `DiscordService.UpdateStatus` method to wrap this.